monai/transforms/utility/array.py

```diff
@@ -1715,3 +1715,101 @@ def __call__(self, img: NdarrayOrTensor, meta_dict: Mapping | None = None) -> Nd
         if self._do_transform:
             img = self.filter(img)
         return img
+
+
+class CoordinateTransform(InvertibleTransform, Transform):
+    """
+    Transform points between image coordinates and world coordinates.
+
+    Args:
+        dtype: The desired data type for the output.
+        mode: Specifies the direction of transformation. This should be an instance of
+            :py:class:`CoordinateTransformMode` enum, with either 'IMAGE_TO_WORLD' or 'WORLD_TO_IMAGE' as the value.
+        affine_lps_to_ras: Defaults to ``False``. Set this to ``True`` if: 1) The image is read by ITKReader,
+            and 2) The ITKReader has `affine_lps_to_ras=True`, and 3) The data is in world coordinates.
+            This ensures that the affine transformation between LPS (left-posterior-superior) and RAS
+            (right-anterior-superior) coordinate systems is correctly applied.
+    """
+
+    def __init__(
+        self,
+        dtype: DtypeLike = torch.float64,
+        mode=CoordinateTransformMode.IMAGE_TO_WORLD,
+        affine_lps_to_ras: bool = False,
+    ) -> None:
+        self.dtype = dtype
+        self.mode = mode
+        self.affine_lps_to_ras = affine_lps_to_ras
+
+
+    @staticmethod
+    def apply_affine_to_points(data, affine, dtype):
+        data = convert_to_tensor(data, track_meta=get_track_meta())
+        data_: torch.Tensor = convert_to_tensor(data, track_meta=False, dtype=dtype)
+
+        homogeneous = concatenate((data_[0], torch.ones((data_[0].shape[0], 1))), axis=1)
+        transformed_homogeneous = torch.matmul(affine, homogeneous.T)
+        transformed_coordinates = transformed_homogeneous[:-1].T.unsqueeze(0)
+        out, *_ = convert_to_dst_type(transformed_coordinates, data, dtype=dtype)
+
+        return out
+
+    def transform_coordinates(self, data, i2w_affine):
+        """
+        Transform coordinates using an affine transformation matrix.
+
+        Args:
+            data: The input coordinates, assume to be in shape (C, N, 3 or 4).
```

Comment:
Hi @KumoLiu , thank you for the PR.

The `transform_coordinates` method assumes that the input `data` has shape `(C, N, 3 or 4)`, but does not validate this. Consider adding input validation to ensure correct usage.

Reply:
Add shape check here, thanks.